std/experimental/allocator/gc_allocator.d

```diff
@@ -140,11 +141,15 @@ struct GCAllocator
     import std.typecons : Ternary;
 
     // test allocation sizes
-    assert(GCAllocator.instance.goodAllocSize(1) == 16);
+    () nothrow @safe @nogc {
+        assert(GCAllocator.instance.goodAllocSize(1) == 16);
+    }();
```

Comment:
same here and everywhere, i.e. we want assert to be top-level to make sure all code goes away in release mode

Reply:
Why? This is inside a unittest after all. Unittests should be able to verify that code built with optimizations is also correct.